tools/slang-test/options.h

```diff
@@ -0,0 +1,83 @@
+// options.h
+
+#ifndef OPTIONS_H_INCLUDED
+#define OPTIONS_H_INCLUDED
+
+#include "../../source/core/dictionary.h"
+
+#include "test-reporter.h"
+#include "render-api-util.h"
+#include "../../source/core/smart-pointer.h"
+
+// A category that a test can be tagged with
+struct TestCategory: public Slang::RefObject
+{
+    // The name of the category, from the user perspective
+    Slang::String name;
+
+    // The logical "super-category" of this category
+    TestCategory* parent;
+};
+
+struct TestCategorySet
+{
+public:
+        /// Find a category with the specified name. Returns nullptr if not found
+    TestCategory * find(Slang::String const& name);
+        /// Adds a category with the specified name, and parent. Returns the category object.
+        /// Parent can be nullptr
+    TestCategory* add(Slang::String const& name, TestCategory* parent);
+        /// Finds a category by name, else reports and writes an error  
+    TestCategory* findOrError(Slang::String const& name);
+
+    Slang::RefPtr<TestCategory> defaultCategory;    ///< The default category
+
+protected:
+    Slang::Dictionary<Slang::String, Slang::RefPtr<TestCategory> > m_categoryMap;
+};
+
+struct Options
+{
+    char const* appName = "slang-test";
+
+    // Directory to use when looking for binaries to run
+    char const* binDir = "";
+
+    // only run test cases with names that have this prefix
+    char const* testPrefix = nullptr;
+
+    // generate extra output (notably: command lines we run)
+    bool shouldBeVerbose = false;
+
+    // force generation of baselines for HLSL tests
+    bool generateHLSLBaselines = false;
```

Comment:
Seeing this line of code again (you didn't add it, but because it moved it shows up in the diffs) makes me realize that we have a lot of little pieces of garbage left in this testing framework. I have *never* used this flag (to my knowledge), so it is purely vestigial.